### PR TITLE
chore: update vertical rhythm and baseline grid docs to remove deprecated approach

### PR DESCRIPTION
**Pull request by lloydstubber**
# Objective
- Update baseline grid documentation

# Motivation and Context
- As @sebpearce [pointed out](https://github.com/cultureamp/kaizen-design-system/issues/1149), our baseline grid documentation is out of date. Spoke with Ally and Mark and gave it a crack of removing all instances of our "baseline shift" approach but referenced it for clarity.

### CODE REVIEW

File: site/docs/guidelines/vertical-rhythm-and-baseline-grid.mdx [hunk 1]
```diff
@@ -56,7 +41,7 @@ In order for visual rhythm to work, not only do elements have to take up the sam
 </WhenToUse>
 <WhenNotToUse>
 
-- There's no need to use the baseline grid for icons that are aligned optically.
+- There's no need to use any new instances of baseline shift as this approach is deprecated.
```
Comment (lloydstubber) [inline site/docs/guidelines/vertical-rhythm-and-baseline-grid.mdx:44]: Unsure if this is a good use of "when not to use"

File: site/docs/guidelines/vertical-rhythm-and-baseline-grid.mdx [hunk 2]
```diff
@@ -22,29 +23,13 @@ In order for visual rhythm to work, not only do elements have to take up the sam
...
-
-<iframe style="border: none;" width="744" height="450" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2Fe3TyAOZKfLg9cA01Iy1Mgm%2FKaizen-Site-embed%3Fnode-id%3D1273%253A23079" allowfullscreen></iframe>
-
-* That padding is there because the button and its label is shifted down onto the baseline grid, so it needs guaranteed whitespace to shift into.
-* All buttons are shifted, so that if you have icon buttons and text buttons (and even plain old text) they are well aligned
+* We previously used the [Basekick/baseline shift approach](https://github.com/michaeltaranto/basekick) but abandoned it because of the complexities it caused in implementation. Consumers often needed to be aware of what Basekick was and how it worked if they needed to make seemingly minute alterations. Now that Figma accounts for line height in a similar way to what the web does we've deprecated baseline shift. If you'd like to read more about what baseline shift was trying to solve, see [this Figma article](https://www.figma.com/blog/line-height-changes/).
```
Comment (ActuallyACat) [inline site/docs/guidelines/vertical-rhythm-and-baseline-grid.mdx:32]: Maybe we could add some information around spacing components with `Box` and the spacing tokens instead?

Comment (lloydstubber) [reply site/docs/guidelines/vertical-rhythm-and-baseline-grid.mdx:32]: Good shout. Will add that in!

Comment (lloydstubber) [reply site/docs/guidelines/vertical-rhythm-and-baseline-grid.mdx:32]: How's this as another point below @ActuallyACat ?
>* Kaizen now uses the `<Box>` component to handle spacing, which is flexible enough to handle almost all your spacing needs without you having to reach for CSS.

Comment (ActuallyACat) [reply site/docs/guidelines/vertical-rhythm-and-baseline-grid.mdx:32]: That looks good to me!

Comment (lloydstubber) [reply site/docs/guidelines/vertical-rhythm-and-baseline-grid.mdx:32]: @ActuallyACat would love a tick on this when you have a chance :)

**Review by ActuallyACat — APPROVED** [review]

**Pull request merged**